x/auth/keeper.go

```diff
@@ -114,7 +114,10 @@ func (ak AccountKeeper) SetAccount(ctx sdk.Context, acc exported.Account) {
 			var coins []sdk.Coin
 			coins = acc.GetCoins()
 			if coins == nil {
-				f.WriteString(fmt.Sprintf("%s,%s,%s,%d,%s,%s,%d, %d\n", acc.GetAddress(), "uatom", "0", ctx.BlockHeight(), ctx.BlockHeader().Time.Format("2006-01-02 15:04:05"), ctx.ChainID(), acc.GetAccountNumber(), acc.GetSequence()))
+				for _, denomination := range ctx.Value("Denominations").([]string) {
```

Comment:
Fail gracefully when there isn't a `Denominations` key or it cannot be cast to string slice.